src/shader_preprocessor.ts

```diff
@@ -339,16 +342,24 @@ function generateGetOutputCoords(
     }
   }
 
-  const dimensions = [];
-  for (let i = 0; i < rank; i++) {
-    dimensions.push(`d${i}`);
-  }
+  let snippet;
+  if (isScalar === true) {
+    snippet = `int getOutputCoords() {
+      return 0;
+    }`;
+  } else {
+    const dimensions = [];
+    for (let i = 0; i < rank; i++) {
+      dimensions.push(`d${i}`);
+    }
+
+    const dtype = getCoordsDataType(rank);
 
-  const dtype = getCoordsDataType(rank);
+    snippet = `${dtype} getOutputCoords() {
+      ${gatherDimensionsStr}
+      return ${dtype}(${dimensions.join(',')});
+    }`;
+  }
 
-  const snippet = `${dtype} getOutputCoords() {
-    ${gatherDimensionsStr}
-    return ${dtype}(${dimensions.join(',')});
-  }`;
```

Comment:
How about just changing it as below? You can't assume that the return type is int.
```
if (dimensions.length === 0)
{
  const snippet = `${dtype} getOutputCoords() {	
    ${gatherDimensionsStr}	
    return ${dtype}(0);	
  }`;
}
else
{
  const snippet = `${dtype} getOutputCoords() {	
    ${gatherDimensionsStr}	
    return ${dtype}(${dimensions.join(',')});	
  }`;
}